util/network-devp2p/src/discovery.rs

```diff
@@ -565,31 +625,41 @@ impl<'a> Discovery<'a> {
 			}
 		};
 
-		if self.is_a_valid_known_node(&node) {
-			self.respond_with_discovery(target, &node)?;
-		} else {
+		match self.check_validity(&node) {
+			NodeValidity::Ourselves => (), // It makes no sense to respond to the discovery request from ourselves
+			NodeValidity::ValidNode(_) => self.respond_with_discovery(target, &node)?,
 			// Make sure the request source is actually there and responds to pings before actually responding
-			self.try_ping(node, PingReason::FromDiscoveryRequest(target));
+			invalidity_reason => self.try_ping(node, PingReason::FromDiscoveryRequest(target, invalidity_reason))
```

Comment:
It can be either `ExpiredNode` or `UnknownNode`.
I can enumerate, true — but will it actually improve readability here, given that we also need to name the value in the pattern match anyway, since we consume it for the `PingReason`

Reply:
It is only for full text search of a variant (If I want to know what is done on 'ExpiredNode' I got the info directly, but it is true that searching on the Enum name could be a better practice). But yes it does only make sense if the rule is applied everywhere.